packages/rs-drive-abci/tests/strategy_tests/main.rs

```diff
@@ -2191,14 +2197,400 @@ mod tests {
+                        status: AssetUnlockStatus::Unknown,
+                    },
+                );
+            });
+
+            outcome
+        };
+
+        // Run block 4
+        // Should change pooled status to broadcasted
+        let mut broadcasted_transactions_amount = last_block_withdrawals.len();
+        let ChainExecutionOutcome {
+            abci_app,
+            proposers,
+            quorums,
+            current_quorum_hash,
+            current_proposer_versions,
+            end_time_ms,
+            withdrawals: last_block_withdrawals,
+            ..
+        } = {
+            let outcome = continue_chain_for_strategy(
+                abci_app,
+                ChainExecutionParameters {
+                    block_start: 4,
+                    core_height_start: 1,
+                    block_count: 1,
+                    proposers,
+                    quorums,
+                    current_quorum_hash,
+                    current_proposer_versions: Some(current_proposer_versions),
+                    start_time_ms: GENESIS_TIME_MS,
+                    current_time_ms: end_time_ms + 1000,
+                },
+                strategy.clone(),
+                config.clone(),
+                StrategyRandomness::SeedEntropy(3),
+            );
+
+            let withdrawal_documents_pooled = outcome
+                .abci_app
+                .platform
+                .drive
+                .fetch_withdrawal_documents_by_status(
+                    withdrawals_contract::WithdrawalStatus::POOLED.into(),
+                    None,
+                    platform_version,
+                )
+                .unwrap();
+
+            let withdrawal_documents_broadcasted = outcome
+                .abci_app
+                .platform
+                .drive
+                .fetch_withdrawal_documents_by_status(
+                    withdrawals_contract::WithdrawalStatus::BROADCASTED.into(),
+                    None,
+                    platform_version,
+                )
+                .unwrap();
+
+            // Ensure total amount of withdrawal documents in broadcasted state is correct
+            assert_eq!(
+                withdrawal_documents_broadcasted.len(),
+                broadcasted_transactions_amount + outcome.withdrawals.len()
+            );
+
+            // Simulate transactions being added to the core mempool
+            let mut core_state = shared_core_state.lock().unwrap();
+            outcome.withdrawals.iter().for_each(|(index, _)| {
+                core_state.asset_unlock_statuses.insert(
+                    *index,
+                    AssetUnlockStatusResult {
+                        index: *index,
+                        status: AssetUnlockStatus::Unknown,
+                    },
+                );
+            });
+
+            outcome
+        };
+
+        // TODO: Do more Core blocks (>48) after broadcasting to see expired transactions
```

Comment:
Produce more Core blocks (>48) after broadcasting to see expired transactions and do not include some of them in the Core so we will have expired withdrawals